builder/dockerfile/copy.go

```diff
@@ -305,10 +305,14 @@ func downloadSource(output io.Writer, stdout io.Writer, srcURL string) (remote b
 	if err != nil {
 		return
 	}
-	filename := filepath.Base(filepath.FromSlash(u.Path)) // Ensure in platform semantics
-	if filename == "" {
-		err = errors.Errorf("cannot determine filename from url: %s", u)
-		return
+	// Use index.html if path is "" or "/" (e.g., http://example.com or http://example.com/)
+	filename := "index.html"
```

Comment:
Thinking;

- first of all, we can't assume `.html` here (could be any kind of file)
- server can send a [`Content-Disposition: attachment; filename="<something>"`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) - do we handle those currently?

If no filename can be determined - perhaps we _can_ generate a filename (could be just `unnamed-file` - shouldn't matter, because it's only the temp-file), however _only_ support this if there's a target filename set as well, i.e.;

```
ADD https://example.com/ filename.foo
```

And fail if;

```
ADD https://example.com/ /some/path/
```

Challenge here may be to determine if the target is a _filename_ or a _path_ (which cannot be determined if the target doesn't exist), i.e.

```
ADD https://example.com/ filename.foo
```

Here, `filename.foo` could mean;

- create a _directory_ named `filename.foo`, and copy the downloaded file into it
- save the downloaded file as `filename.foo`

(haven't checked the current semantics on this)